src/read.js

```diff
@@ -61,17 +61,31 @@ export async function parquetRead(options) {
 
   // onComplete transpose column chunks to rows
   if (onComplete) {
-    /** @type {any[][]} */
-    const rows = []
-    for (const asyncGroup of assembled) {
-      // filter to rows in range
-      const selectStart = Math.max(rowStart - asyncGroup.groupStart, 0)
-      const selectEnd = Math.min((rowEnd ?? Infinity) - asyncGroup.groupStart, asyncGroup.groupRows)
-      // transpose column chunks to rows in output
-      const groupData = await asyncGroupToRows(asyncGroup, selectStart, selectEnd, columns, rowFormat)
-      concat(rows, groupData.slice(selectStart, selectEnd))
+    if (rowFormat === 'object') {
+      /** @type {Record<string, any>[]} */
+      const rows = []
+      for (const asyncGroup of assembled) {
+        // filter to rows in range
+        const selectStart = Math.max(rowStart - asyncGroup.groupStart, 0)
+        const selectEnd = Math.min((rowEnd ?? Infinity) - asyncGroup.groupStart, asyncGroup.groupRows)
+        // transpose column chunks to rows in output
+        const groupData = await asyncGroupToRows(asyncGroup, selectStart, selectEnd, columns, rowFormat)
+        concat(rows, groupData.slice(selectStart, selectEnd))
+      }
+      onComplete(rows)
+    } else {
+      /** @type {any[][]} */
+      const rows = []
+      for (const asyncGroup of assembled) {
+        // filter to rows in range
+        const selectStart = Math.max(rowStart - asyncGroup.groupStart, 0)
+        const selectEnd = Math.min((rowEnd ?? Infinity) - asyncGroup.groupStart, asyncGroup.groupRows)
+        // transpose column chunks to rows in output
+        const groupData = await asyncGroupToRows(asyncGroup, selectStart, selectEnd, columns, rowFormat)
+        concat(rows, groupData.slice(selectStart, selectEnd))
+      }
```

Comment:
These branches appear identical except the type? Can we avoid this duplication somehow? Even just a type assertion would be better than duplication in my opinion.

Reply:
I expected typescript would infer automatically which overload to use depending on the value of `rowFormat`. I'm not sure why it does not.